src/manifest/representation.ts

```diff
@@ -111,6 +111,14 @@ class Representation {
    */
   public codec : string | undefined;
 
+  /**
+   * Supplemental codecs are defined as backwards-compatible codecs enhancing
+   * the experience of a base layer codec
+   */
+  public supplementalCodec?: string | undefined;
```

Comment:
Note that as the `codec` property will keep the original value which might not be the one actually relied on, we may have other parts of the code relying on this property directly where they perhaps meant to use `supplementalCodec` instead.

Basically this PR gives a new property, `supplementalCodec`, the role of `codec` in some situations, with `codec` consequently ignored in those, so we also have to study if those other usages of the `codec` have to be replaced.

Reply:
I'm here thinking particularly of the API, for example `getVideoTrack` will return as a `codec` property the `Representation`'s codec even if we're actually relying on the `supplementalCodec` for SourceBuffer creation and support testing.
I'm not sure of what we should do here:
  1. have both properties also in the API, in which case we kind of change the API behavior (it could still easily be argued that this is not a breaking change) and move some complexity to the application (yet as an advantage, the application can have the most information possible)
  2. having the API's `codec` reflect the actual codec used for support testing and SourceBuffer creation, so equal to a `Representation`'s `supplementalCodec` in some cases.
  3. Keeping the API's `codec` as is, with the application not knowing our internal logic. The sad thing here is that it wouldn't be able to tel that we're playing e.g. Dolby Vision when we rely on supplementalCodec for it

Reply:
Thoughts?

There may even be more `codec` usage in the code, but that was the one on top of my head.

Reply:
I think both 1 and 2 are fine approach, but I think 2 is the most reliable for an application as you can know without doubt what codec has been used to create the SourceBuffer. 
It keeps the public API simple as it's hide some complexity.
One drawback is that this API will not return the base codec if the supplemental codec has been used instead,
but I doubt that it is an information you want to have.

Reply:
Yeah, I'm also wondering if there's scenario where both `supplementalCodec` and `codec` are important, as the wording of the HLS' `SUPPLEMENTAL-CODEC` seem to indicate that this is possible and the SCTE-214 spec most likely took the `supplementalProperty`'s behavior from HLS.

But in that case, it probably would also be important for `addSourceBuffer` and `isTypeSupported` presumably?

It's not so simple without having many examples though. Both the HLS and SCTE214 specs only talks concretely about Dolby Vision for which the `supplementalProperty || codec` logic works.